src/main/scala/com/fulcrumgenomics/bam/ComputeRnaSeqInsertSize.scala

```diff
@@ -0,0 +1,362 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import java.io.{Closeable, File, InputStream}
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.bam.GeneAnnotations._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util._
+import dagr.commons.CommonsDef.{PathToBam, PathToFastq}
+import dagr.commons.util.LazyLogging
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.SamPairUtil.PairOrientation
+import htsjdk.samtools._
+import htsjdk.samtools.filter._
+import htsjdk.samtools.util.{CoordMath, Interval, OverlapDetector}
+
+import scala.collection.JavaConversions.iterableAsScalaIterable
+import scala.io.Source
+
+@clp(description =
+  """Computes the insert size for RNA-Seq experiments.
+    |
+    |Computes the insert size by counting the # of bases sequenced in transcript space.  The insert size is defined
+    |as the distance between the first bases sequenced of each pair respectively (5' sequencing ends).
+    |
+    |This tool skips reads that overlap multiple genes, reads that aren't fully enclosed in a gene, and reads where the
+    |insert size would disagree across transcripts from the same gene.  Also skips reads that are unpaired, failed QC,
+    |secondary, supplementary, pairs without both ends mapped, duplicates, and pairs whose reads map to different
+    |chromosomes. Finally, skips transcripts where the too few mapped read bases overlap exonic sequence.
+    |
+    |This tool requires each mapped pair to have the mate cigar (MC) tag and for read pairs not to overlap.  Use
+    |SetMateInformation to add the former or Fgbio's ClipOverlappingReads to add both.
+  """,
+  group = ClpGroups.SamOrBam)
+class ComputeRnaSeqInsertSize
+(@arg(flag="i", doc="Input BAM file.") val input: PathToBam,
+ @arg(flag="r", doc="Input gene annotations in refFlat form.  Format described here: http://genome.ucsc.edu/goldenPath/gbdDescriptionsOld.html#RefFlat")
+ val refFlat: FilePath,
+ @arg(flag="o", doc="Output metrics file to which to write.") val output: FilePath,
```

Comment:
Also making it optional, and use the input if not given (sans the input's file extension)